bigbluebutton-html5/imports/ui/components/debug-window/component.jsx

```diff
@@ -0,0 +1,214 @@
+import React, { PureComponent } from 'react';
+import cx from 'classnames';
+import Draggable from 'react-draggable';
+import {
+  defineMessages, injectIntl, intlShape, FormattedMessage,
+} from 'react-intl';
+import styles from './styles.scss';
+import Modal from '/imports/ui/components/modal/simple/component';
+import { withModalMounter } from '/imports/ui/components/modal/service';
+import Toggle from '/imports/ui/components/switch/component';
+import Button from '/imports/ui/components/button/component';
+import Storage from '/imports/ui/services/storage/session';
+import { withLayoutConsumer } from '/imports/ui/components/layout/context';
+
+const SHOW_DEBUG_WINDOW = Meteor.settings.public.app.enableDebugWindowButton;
+const SHOW_DEBUG_WINDOW_SHORTCUT = Meteor.settings.public.app.enableDebugWindowShortcut;
+const SHOW_DEBUG_WINDOW_ACCESSKEY = Meteor.settings.public.app.shortcuts.openDebugWindow.accesskey;
+
+const messages = defineMessages({
+  debugModalTitle: {
+    id: 'app.debugWindow.windowTitle',
+    description: 'Debug window title',
+  },
+  userAgentLabel: {
+    id: 'app.debugWindow.form.userAgentLabel',
+    description: 'User agent form label',
+  },
+  copyButtonLabel: {
+    id: 'app.debugWindow.form.button.copy',
+    description: 'User agent form copy button',
+  },
+  enableAutoarrangeLayoutLabel: {
+    id: 'app.debugWindow.form.enableAutoarrangeLayoutLabel',
+    description: 'Enable Autoarrange layout label',
+  },
+  enableAutoarrangeLayoutDescription: {
+    id: 'app.debugWindow.form.enableAutoarrangeLayoutDescription',
+    description: 'Enable Autoarrange layout description',
+  },
+});
+
+class DebugWindow extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dragging: false,
+      position: {
+        x: 0,
+        y: 0,
+      },
+    };
+
+    this.handleResize = this.handleResize.bind(this);
+    this.handleOnDrag = this.handleOnDrag.bind(this);
+    this.handleDragStop = this.handleDragStop.bind(this);
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleResize, false);
+    document.addEventListener('keyup', (event) => {
```

Comment:
I don't think we need to add another event listener. The button on line `191` just needs `accessKey={SHOW_DEBUG_WINDOW_ACCESSKEY}` added to it as an attribute.

Reply:
I added a setting to disable the button, then the button not be render and the accessKey not be work

Reply:
Ahh, so the idea was to have the shortcut still functional when the button is hidden.